### PR TITLE
FragmentDisplayContext.java

### PR DESCRIPTION
**Pull request by jkappler**

### CODE REVIEW

**Comment by p2kmgcl** [comment]
@carloslancha I have applied the changes you told me, please can you check if everything is fine? :heart:

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentEditor.es.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	 * Necesary for getting the real inputs which interact with the server.
+	 * @type {string}
+	 */
+	namespace: Config.string().required(),
+
+	/**
+	 * Initial HTML sent to the editor
+	 * @type {string}
+	 */
+	initialHTML: Config.string().required(),
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentEditor.es.js]: `.writeOnce()`?

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentEditor.es.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+	/**
+	 * Initial CSS sent to the editor
+	 * @type {string}
+	 */
+	initialCSS: Config.string().required(),
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentEditor.es.js]: `.writeOnce()`?

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentEditor.es.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	/**
+	 * Initial JS sent to the editor
+	 * @type {string}
+	 */
+	initialJS: Config.string().required(),
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/FragmentEditor.es.js]: `.writeOnce()`?

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditor.soy [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * SourceEditor
+ */
+{template .render}
+	{@param syntax: string}
+	{@param toolbarItems: list<string>}
+	{@param? aceEditor_: any}
+	{@param? _handleEditorReady: any}
+	{@param? _handleChangeContent: any}
+
+	<div class="source-editor" ref="wrapper">
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditor.soy]: If ref is not used is better not to specify it.

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditor.es.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import {Config} from 'metal-state';
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditor.es.js]: Everything should be alphabetically ordered

File: modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditorToolbar.es.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	/** @inheritdoc */
+	rendered() {
+		if (this.showToolbar && this.aceEditor) {
+			this.items.forEach((itemName) => {
+				const ItemComponent = ITEM_MAP[itemName];
+				const itemWrapper = this.refs[itemName];
+
+				new ItemComponent({
```
Comment (carloslancha) [inline modules/apps/web-experience/modern-site-building/modern-site-building-fragment-web/src/main/resources/META-INF/resources/js/SourceEditorToolbar.es.js]: Can't we render this from .soy? If rendered here I guess we need to save the instances and destroy'em on component dispose, which is done by default if instantiated from .soy

**Pull request closed without merging**